centreon/src/Core/Security/ProviderConfiguration/Domain/SecurityAccess/RolesMapping.php

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Configured conditions must be satisfied to be authorized and map IDP's roles and Centreon's ACLs
+ *
+ * @see RolesMapping::validate()
+ */
+class RolesMapping implements SecurityAccessInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @var string
+     */
+    private string $scope = 'undefined';
+
+    /**
+     * @param LoginLoggerInterface $loginLogger
+     * @param AttributePathFetcher $attributePathFetcher
+     */
+    public function __construct(
+        private readonly LoginLoggerInterface $loginLogger,
+        private readonly AttributePathFetcher $attributePathFetcher
+    ) {
+    }
+
+    /**
+     * @param string $token
+     * @param Configuration $configuration
+     * @return void
+     * @throws AclConditionsException
+     */
+    public function validate(string $token, Configuration $configuration): void
+    {
+        $this->scope = $configuration->getType();
+        $customConfiguration = $configuration->getCustomConfiguration();
+        $aclConditions = $customConfiguration->getACLConditions();
+        if (!$aclConditions->isEnabled()) {
+            $this->loginLogger->info($this->scope, "Roles mapping is disabled");
+            $this->info("Roles mapping is disabled");
+            return;
+        }
+
+        $this->loginLogger->info($this->scope, "Roles mapping is enabled");
+        $this->info("Roles mapping is enabled");
+
+        $identityProviderIntrospectionData = $this->attributePathFetcher->fetch(
+            $token,
+            $configuration,
+            $aclConditions->getEndpoint()
+        );
+
+        $attributePath = explode(".", $aclConditions->getAttributePath());
+        foreach ($attributePath as $attribute) {
+            $providerConditions = [];
+            if (array_key_exists($attribute, $identityProviderIntrospectionData)) {
+                $providerConditions = $identityProviderIntrospectionData[$attribute];
+                $identityProviderIntrospectionData = $identityProviderIntrospectionData[$attribute];
+            } else {
+                break;
+            }
+        }
+
+        $configuredClaimValues = $aclConditions->getClaimValues();
+        if ($aclConditions->onlyFirstRoleIsApplied() && !empty($configuredClaimValues)) {
+            $configuredClaimValues = [$configuredClaimValues[0]];
+        }
+
+        if (is_string($providerConditions)) {
+            $providerConditions = explode(",", $providerConditions);
+        }
+
+        $this->validateAclAttributeOrFail($providerConditions, $configuredClaimValues);
+    }
+
+    /**
+     * Validate roles mapping conditions
+     *
+     * @param array<mixed> $conditions
+     * @param string[] $configuredAuthorizedValues
+     * @throws AclConditionsException
+     */
+    private function validateAclAttributeOrFail(array $conditions, array $configuredAuthorizedValues): void
+    {
+        if (array_is_list($conditions) === false) {
```

Comment:
The function has a hard bool return type.
The equal to false is weird to me.
```suggestion
        if (! array_is_list($conditions)) {
```